### ISSUE TITLE
living target for ECS

### ISSUE THREAD

**Issue by neu5ron** [rocknsm/rock-dashboards#46]
## Logstash
- [ ] Document or note new ECS fields
  - [x] `TLS` 
  - [x] `Threat.`
  - [ ] `Rule.`
  - [ ] `DNS.`
    - [ ] `sub_domain`
    - [ ] `top_level_domain`
  - [ ] `Domain.`
    - [ ] `sub_domain`
    - [ ] `top_level_domain`
  - [ ] `Package`
    - [ ] pending confirmation based on whether only should be "installed" software, or can include just metadata of extracted/seen/passive software information
- [ ] Document or note new ECS fields for Zeek
  - [x] TLS
    - [x] ssl.log
    - [x] no other logs that seem to apply, maybe a field or two in rdp.log or smtp.log but they i would say no...
  - [ ] notice.log to rule.name
- [ ] Document or note new Suricata
  - [ ] TLS
    - [ ] tls log
      - [ ] generate logs with extended/custom
   - [ ] alert log to rule schema
- [ ] Update logstash config for Zeek
  - [ ] TLS 
- [ ] Update logstash config for Suricata
  - [ ] TLS 

## Kibana

- [ ] Update migration dashboards for Zeek
- [ ] Update migration dashboards for Suricata
- [ ] Tweak docket pivot
- [ ] Check any other dashboards
- [ ] Test / Save everything for 7.5 objects

**Comment by ipninichuck** [comment]
I have a question about the alert( ex: alert.action) category that is being used. Should we be swapping this for the rule category? It would seem that with a little expansion the rule category could encompass all info relating to alerts triggered. Though it is interesting to note that the filebeat-* index has not done so with their Zeek or Suricata modules yet, though I suspect it might be coming.

**Comment by peasead** [comment]
> I have a question about the alert( ex: alert.action) category that is being used. Should we be swapping this for the rule category? It would seem that with a little expansion the rule category could encompass all info relating to alerts triggered. Though it is interesting to note that the filebeat-* index has not done so with their Zeek or Suricata modules yet, though I suspect it might be coming.

I'll take a stab at this. I apologize if I'm missing your question.

`filebeat-*` does have `suricata.eve.alert.category` and `suricata.eve.alert.action` as Exported Fields ([Note](https://www.elastic.co/guide/en/beats/filebeat/master/exported-fields-suricata.html)). I'm not sure it would make sense combined or expanded as they're providing different information, as you can see below. 

`suricata.eve.alert.action` would be ([Note](https://suricata.readthedocs.io/en/latest/configuration/suricata-yaml.html#suricata-yaml-action-order)):
• Pass
• Drop
• Reject
• Alert

`suricata.eve.alert.category` would be:
• PHISHING
• COINMINER
• JA3
• EXPLOIT_KIT
• HUNTING
• ADWARE_PUP
• MALWARE
• ...

I'm unfamiliar with `rule.*` in the Suricata module, though I'd assume it's `suricata.eve.alert.*`.

I'm not quite sure `alert` would make sense in a Zeek module though, outside of _maybe_ the [Intel Framework](https://www.elastic.co/guide/en/beats/filebeat/master/exported-fields-zeek.html#_intel), but I'm not sure that is really an alert in the traditional sense. ([Note](https://www.elastic.co/guide/en/beats/filebeat/master/exported-fields-zeek.html)).

Please correct me if I missed what you were asking.

**Comment by neu5ron** [comment]
yup, alert will be included. thanks for bringing it up! Im not sure the exact format of the fields yet - but im adding to the list of applicable fields

**Comment by ipninichuck** [comment]
I also noticed those fields in the filebeat modules. That is why I was
confused when the ECS 1.4 docs had no reference to the categories. When I
brought it up on the Elastic Slack someone brought up the new rules
category. It will be interesting to see how it plays out. For now I'm
highlighting the fields in my spreadsheet as possibly changing. Than you
for the further information.
